controllers/binding_controller.go

```diff
@@ -54,12 +55,31 @@ func (r *BindingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return reconcile.Result{}, client.IgnoreNotFound(err)
 	}
 
-	rabbitClient, err := r.RabbitmqClientFactory(ctx, r.Client, binding.Spec.RabbitmqClusterReference, binding.Namespace)
+	systemCertPool, err := x509.SystemCertPool()
+	if err != nil {
+		msg := "failed to retrieve system trusted certs"
+		r.Recorder.Event(binding, corev1.EventTypeWarning, "FailedUpdate", msg)
+		logger.Error(err, msg)
+		return ctrl.Result{}, err
+	}
 
-	if errors.Is(err, internal.NoSuchRabbitmqClusterError) && binding.ObjectMeta.DeletionTimestamp.IsZero() {
+	rmq, svc, secret, err := internal.ParseRabbitmqClusterReference(ctx, r.Client, binding.Spec.RabbitmqClusterReference, binding.Namespace)
+	if errors.Is(err, internal.NoSuchRabbitmqClusterError) && !binding.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.Info(noSuchRabbitDeletion, "binding", binding.Name)
+		r.Recorder.Event(binding, corev1.EventTypeNormal, "SuccessfulDelete", "successfully deleted binding")
+		return reconcile.Result{}, r.removeFinalizer(ctx, binding)
+	}
+	if errors.Is(err, internal.NoSuchRabbitmqClusterError) {
```

Comment:
I would keep the comment for this requeue after 10s:
```
// If the object is not being deleted, but the RabbitmqCluster no longer exists, it could be that
// the Cluster is temporarily down. Requeue until it comes back up.
```
what do you think about adding the comment here? is it unnecessary?